app/src/main/java/com/xmartlabs/template/service/gson/RequiredFieldDeserializer.java

```diff
@@ -0,0 +1,50 @@
+package com.xmartlabs.template.service.gson;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParseException;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Type;
+import java.util.Locale;
+
+import timber.log.Timber;
+
+/** {@link JsonDeserializer} that checks that all {@link GsonRequired} fields have a value */
+public class RequiredFieldDeserializer implements JsonDeserializer<Object> {
+  private Gson gson = new Gson();
```

Comment:
Yes, i wanted to discuss this.
There's a limitation in using any injected Gson is that that Gson must not include the `RequiredFieldDeserializer` due to a circular dependency. It makes sense to just add this deserializer to the service Gson only, what do you think?